configuration/config2/EnvironmentNode.cpp

```diff
@@ -190,66 +190,69 @@ std::string EnvironmentNode::getLocalValue() const
 }
 
 
-void EnvironmentNode::validate(Status &status, bool includeChildren) const
+void EnvironmentNode::validate(Status &status, bool includeChildren, bool includeHiddenNodes) const
 {
-    //
-    // Check node value
-    if (m_pLocalValue)
+    if (!m_pSchemaItem->isHidden() || includeHiddenNodes)
     {
-        m_pLocalValue->validate(status, "");
-    }
-
-    //
-    // Check any attributes
-    for (auto attrIt = m_attributes.begin(); attrIt != m_attributes.end(); ++attrIt)
-    {
-        attrIt->second->validate(status, m_id);
+        //
+        // Check node value
+        if (m_pLocalValue)
+        {
+            m_pLocalValue->validate(status, m_id);
+        }
 
         //
-        // If this value must be unique, make sure it is
-        if (attrIt->second->getSchemaValue()->isUniqueValue())
+        // Check any attributes
+        for (auto attrIt = m_attributes.begin(); attrIt != m_attributes.end(); ++attrIt)
         {
-            bool found = false;
-            std::vector<std::string> allValues;
-            attrIt->second->getAllValuesForSiblings(allValues);
-            std::set<std::string> unquieValues;
-            for (auto it = allValues.begin(); it != allValues.end() && !found; ++it)
+            attrIt->second->validate(status, m_id);
+
+            //
+            // If this value must be unique, make sure it is
+            if (attrIt->second->getSchemaValue()->isUniqueValue())
             {
-                auto ret = unquieValues.insert(*it);
-                found = ret.second;
+                bool found = false;
+                std::vector<std::string> allValues;
+                attrIt->second->getAllValuesForSiblings(allValues);
+                std::set<std::string> unquieValues;
+                for (auto it = allValues.begin(); it != allValues.end() && !found; ++it)
+                {
+                    auto ret = unquieValues.insert(*it);
+                    found = ret.second;
+                }
+
+                if (found)
+                {
+                    status.addUniqueMsg(statusMsg::error, m_id, attrIt->second->getName(), "Attribute value must be unique");
```

Comment:
The node ID can easily be used to find the name. The scenario is that the UI would use the nodeID to find a menu item that it currently has cached. That cached item would already have the name. If then node is not cached, then having the name is probably not useful since the UI would need to get the node info anyway.